Dockerfile.dist

```diff
@@ -0,0 +1,31 @@
+# This image will be published as dspace/dspace-angular:$DSPACE_VERSION-dist
+# See https://github.com/DSpace/dspace-angular/tree/main/docker for usage details
+
+# Test build:
+# docker build -f Dockerfile.dist -t dspace/dspace-angular:dspace-7_x-dist .
+
+FROM node:18-alpine as build
+
+# Ensure Python and other build tools are available
+# These are needed to install some node modules, especially on linux/arm64
+RUN apk add --update python3 make g++ && rm -rf /var/cache/apk/*
+
+WORKDIR /app
+COPY package.json yarn.lock ./
```

Comment:
It might be useful to add a few more inline comments to this file just to describe the logic here. For instance, I found it a bit odd that you only copy over the local package.json and yarn.lock here... but I'm guessing it's simply to let the "yarn install" work on the next line?

(I had just found it slightly odd that package.json/yarn.lock were copied when just below you do an `ADD . /app/` to add everything.  But, I'm OK with this approach, and maybe it's even a bit more straightforward... just thought it might be worth describing in comments.)

Reply:
@mark-cooper answered this in a meeting between us today.  This initial COPY is just to provide the few files need for the "yarn install".  It helps streamline the Docker build process by taking advantage of caching between steps.